File: Languages/typescript/src/start.ts
import {Table} from './model/Table'
import { UserInfo } from './model/UserInfo';
import { SimpleTableAnalysis } from './model/SimpleTableAnalysis';
import { MaxValueAnalysis } from './provider/MaxValueAnalysis';
import { MinValueAnalysis } from './provider/MinValueAnalysis';
import { MeanAnalysis } from './provider/MeanAnalysis';

export class Start {

    public constructor(args: string[]) {
        let fileName = this.getParam(args)
        // Obtendo o tempo inicial de leitura em milissegundos
        let leitura_antes = new Date().getTime()

        // Convertendo arquivo em lista de "UserInfo"
        var table: Table = new Table(fileName);

        // Obtendo o tempo final de leitura em milissegundos
        let leitura_depois = new Date().getTime()

        let list: UserInfo[] = table.getUserInfoList();

        let maxAnalysis: SimpleTableAnalysis = new MaxValueAnalysis()
        let minAnalysis: SimpleTableAnalysis = new MinValueAnalysis()
        let meanAnalysis: SimpleTableAnalysis = new MeanAnalysis()

        // Obtendo o tempo inicial de analise em milissegundos
        let analise_antes = new Date().getTime()

        // Realizando analises
        let max = maxAnalysis.analysis(list)
        let min = minAnalysis.analysis(list)
        let mean = meanAnalysis.analysis(list)

        // Obtendo o tempo final de analise em milissegundos
        let analise_depois = new Date().getTime()

        // Dados de saida
        console.log("[OK]Arquivo: " + fileName)
        console.log("[OK]Tempo_leitura: " + (leitura_depois - leitura_antes) + " ms")
        console.log("[OK]Tempo_analise: " + (analise_depois - analise_antes) + " ms")
        console.log("[OK]Max: " + max)
        console.log("[OK]Min: " + min)
        console.log("[OK]Mean: " + mean)
    }

    // Método para captura e tratamento dos parametros obtidos via console
    // @param codes Lista de parametros obtidos via console
    // @return Tamanho de usuários á serem gerados
    private getParam(codes: string[]): string{
        if(codes.length != 3) {
            console.log("Parametros inválidos.")
            process.exit(-1)
        }
    
        return(codes[2])    
    }
}

new Start(process.argv);

File: Languages/kotlin/src/start.kt
import model.UserInfo
import java.util.ArrayList

/**
 * Função de inicialização do projeto
 * @param args Lista de parametros obtidos via console
 */
fun main(args: Array<String>) {
    val tamanho = prepareArgs(args)

    var builder = StringBuilder()
    val list = ArrayList<UserInfo>()

    val antes = System.currentTimeMillis()

    for (index in 0..tamanho - 1) {
        builder.append("user").append(index)
        val user = builder.toString()

        builder = StringBuilder()

        builder.append("password").append(index)
        val password = builder.toString()

        list.add(UserInfo(user, password))
    }

    val time = System.currentTimeMillis() - antes

    builder = StringBuilder()
    builder.append("[OK]Tamanho: ").append(tamanho).append("\n")
    builder.append("[OK]Tempo: ").append(time).append(" ms")


    println(builder.toString())
}

/**
 * Método para captura e tratamento dos parametros obtidos via console
 * @param codes Lista de parametros obtidos via console
 * *
 * @return Tamanho de usuários á serem gerados
 */
private fun prepareArgs(codes: Array<String>): Int {
    if (codes.size != 1) {
        println("Parametros inválidos.")
        System.exit(-1)
    }

    val line = Integer.parseInt(codes[0])

    if (line <= 0) {
        println("Quantidade de linhas menor que 1.")
        System.exit(-1)
    }

    return line
}